### PR TITLE
[CDAP-16338] Moves the polling logic to client

### PR DESCRIPTION
**Pull request by ajainarayanan**
JIRA: https://issues.cask.co/browse/CDAP-16338
Build: https://builds.cask.co/browse/CDAP-UDUT519
E2E tests: https://builds.cask.co/browse/IT-UPD2122-2

### CODE REVIEW

File: cdap-ui/app/cdap/services/datasource/index.js [hunk 1]
```diff
@@ -230,26 +240,33 @@ export default class Datasource {
...
     };
 
-    this.socketSend('poll-start', generatedResource);
+    this.socketSend('request', generatedResource);
 
     return observable;
   }
 
+  startClientPoll = (resourceId, interval) => {
+    const intervalTimer = setInterval(() => {
```
Comment (elfenheart) [inline cdap-ui/app/cdap/services/datasource/index.js]: There might be a potential issue with using interval. We are queing request every 10 seconds, so if for some reason response takes longer than 10 seconds, the request will keep piling. Would it be better to do this as a timeout on the response callback instead?

Comment (ajainarayanan) [reply cdap-ui/app/cdap/services/datasource/index.js]: Good point. Have updated the PR

File: cdap-ui/server.js [hunk 2]
```diff
@@ -156,9 +156,11 @@ getCDAPConfig()
     });
 
     sockServer.on('connection', function(c) {
+      log.info('[SOCKET OPEN] Connection to client "' + c.id + '" opened');
```
Comment (nitinmotgi) [inline cdap-ui/server.js]: Please change this to debug

File: cdap-ui/server/aggregator.js [hunk 3]
```diff
@@ -449,21 +310,13 @@ function onSocketData(message) {
...
         this.pushConfiguration(r);
         break;
-      case 'poll-start':
-        log.debug('[POLL-START]: (method: ' + r.method + ', id: ' + r.id + ', url: ' + r.url + ')');
-        this.startPolling(r);
-        break;
       case 'request':
         r.startTs = Date.now();
-        log.debug('[REQUEST]: (method: ' + r.method + ', id: ' + r.id + ', url: ' + r.url + ')');
+        log.info('[REQUEST]: (method: ' + r.method + ', id: ' + r.id + ', url: ' + r.url + ')');
```
Comment (nitinmotgi) [inline cdap-ui/server/aggregator.js]: This should be debug or info. This can fill up disk.

File: cdap-ui/server/aggregator.js [hunk 4]
```diff
@@ -474,8 +327,7 @@ function onSocketData(message) {
  * @private onSocketClose
  */
 function onSocketClose() {
-  this.stopPollingAll();
-  this.polledResources = {};
+  log.info('[SOCKET CLOSE] Connection to client "' + this.connection.id + '" closed');
```
Comment (nitinmotgi) [inline cdap-ui/server/aggregator.js]: Same level as connect

**Review by nitinmotgi — CHANGES_REQUESTED** [review]
Few minor commends from me about log messages.

**Review by elfenheart — APPROVED** [review]
👍 LGTM besides @nitinmotgi comments about log levels

**Review by nitinmotgi — APPROVED** [review]

**Comment by ajainarayanan** [comment]
Green build: https://builds.cask.co/browse/CDAP-UDUT519-5

**Pull request merged**